### DISCUSSION TITLE
Vim: subword text object?

### DISCUSSION THREAD

**Discussion by thomasheartman** [zed-industries/zed#22280] [Feature Requests]
I know Zed's Vim lets you navigate by subwords (`vim::NextSubwordStart` and friends), but these appear to only be for navigation, not for changing. 

Is there something I'm missing here or would that be a feature request?

For reference, given the following example, where `|` represents the cursor position:

```
CamelCa|seWord
```

I'd like `diw` to yield `CamelWord`.

(I typically use it most for `ciw`, but this was an easier example).

**Comment by mwac-dev** [comment]
this is pretty old but if anyone is wondering how to do this, I managed to get it somewhat working with :

```  {
    "context": "VimControl && !menu && vim_mode != operator",
    "bindings": {
      "alt-w": "vim::NextSubwordStart",
      "alt-b": "vim::PreviousSubwordStart",
      "alt-e": "vim::NextSubwordEnd",
    },
  },
  {
    "context": "vim_mode == normal",
    "bindings": {
      "v i e": ["workspace::SendKeystrokes", "l alt-b v alt-e"],
      "c i e": ["workspace::SendKeystrokes", "l alt-b v alt-e c"],
      "d i e": ["workspace::SendKeystrokes", "l alt-b v alt-e d"],
    },
  },
```
I use an alt prefix that way "w", "b", and "e" are still normal vim bindings, and in the keystrokes I pass "l" to go left one char first to handle an edge case when you trigger this on the first letter of the word.

**Reply by thomasheartman** [reply]
Unless I'm mistaken, you don't need a workaround for this today. There is the `vim::Subword` binding.

So for instance, I have something like this:

```jsonc
  {
    "context": "vim_operator == a || vim_operator == i",
    "bindings": {
      "w": "vim::Subword",
    },
  },
```

At least that solves the use case I mentioned in the OP. It's been quite a while, so I can only guess that that's what I wanted.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
